client/packages/system/src/Item/context/useUnitVariant.ts

```diff
@@ -1,31 +1,40 @@
 import { useUnitVariantList } from '../api';
 import { create } from 'zustand';
-import { UnitVariantNode } from '@common/types';
+import { UnitVariantNode, VariantNode } from '@common/types';
 import { NumUtils, isEqual } from '@common/utils';
 import { useEffect } from 'react';
 
+type UserSelectedVariants = {
+  [itemId: string]: /* userSelectedVariantId */ string;
+};
 interface UnitState {
   // From back end
   items: {
     [itemId: string]: UnitVariantNode;
   };
+  userSelectedVariants: UserSelectedVariants;
```

Comment:
This structure will be stored in app data and loaded on store change (stored be store)